apps/extension/src/background/ports/permission.ts

```diff
@@ -60,7 +63,18 @@ export async function requestPermission(
     request.transaction.origin.id,
     originData
   )
-  if (origin.permissions === "allow") {
+
+  const config = await configManager.getWithDefault(request.transaction.model)
+  const caller = modelCallers[config.id]
+
+  const maxCost = transactionManager.getMaxCost(
+    request.transaction,
+    caller.defaultOptions.prompt_cost, 
+    caller.defaultOptions.completion_cost,
+    caller.defaultOptions.max_tokens
+  )
+
+  if (origin.limit !== 0 && maxCost !== undefined && (origin.limit === -1 || origin.used + maxCost <= origin.limit) ) {
```

Comment:
this logic might be nice to have in the originManager as well (if that's where `used` is moving to, or originManager if not)

Reply:
Got it. I'll move this to usageManager.